src/lib/JreSetupManager.ts

```diff
@@ -116,27 +116,35 @@ class JreSetupManager extends AsyncCreatable {
 	}
 
 	private async verifyJavaVersion(javaHome: string): Promise<void> {
-		const versionOut = await this.fetchJavaVersion(javaHome);
-
-		// Version output looks like this:
-		// MacOS: "openjdk version "11.0.6" 2020-01-14 LTS\nOpenJDK Runtime Environment Zulu11.37+17-CA (build 11.0.6+10-LTS)\nOpenJDK 64-Bit Server VM Zulu11.37+17-CA (build 11.0.6+10-LTS, mixed mode)\n"
-		// Win10: "openjdk 14 2020-03-17\r\nOpenJDK Runtime Environment (build 14+36-1461)\r\nOpenJDK 64-Bit Server VM (build 14+36-1461, mixed mode, sharing)\r\n"
-		// We want to get the "11.0" or "14" part
-		// The version number could be of the format 11.0 or 1.8 or 14
-		const regex = /(\d+)(\.(\d+))?/;
-		const matchedParts = regex.exec(versionOut);
-		this.logger.trace(`Version output match for pattern ${regex.toString()} is ${JSON.stringify(matchedParts)}`);
-
-		// matchedParts should have four groups: "11.0", "11", ".0", "0" or "14", "14", undefined, undefined
-		if (!matchedParts || matchedParts.length < 4) {
-			throw new SfError(getMessage(BundleName.JreSetupManager, 'VersionNotFound', [this.configFile]));
+		const versionCommandOut = await this.fetchJavaVersion(javaHome);
+
+		// We are using "java -version" below which has output that typically looks like:
```

Comment:
This comment is a little confusingly worded. It looks like we're defaulting to the output expectation for `java -version`, but falling back on `java --version` if no match is found, which makes sense but the comment doesn't make that totally clear.